src/components/Stack/Stack.tsx

```diff
@@ -32,33 +32,31 @@ export interface StackProps {
   distribution?: Distribution;
 }
 
-export class Stack extends React.PureComponent<StackProps, never> {
-  static Item = Item;
-
-  render() {
-    const {
-      children,
-      vertical,
-      spacing,
-      distribution,
-      alignment,
-      wrap,
-    } = this.props;
-
-    const className = classNames(
-      styles.Stack,
-      vertical && styles.vertical,
-      spacing && styles[variationName('spacing', spacing)],
-      distribution && styles[variationName('distribution', distribution)],
-      alignment && styles[variationName('alignment', alignment)],
-      wrap === false && styles.noWrap,
-    );
-
-    const itemMarkup = elementChildren(children).map((child, index) => {
-      const props = {key: index};
-      return wrapWithComponent(child, Item, props);
-    });
-
-    return <div className={className}>{itemMarkup}</div>;
-  }
-}
+export const Stack = memo(function Stack({
+  children,
+  vertical,
+  spacing,
+  distribution,
+  alignment,
+  wrap,
+}: StackProps) {
+  const className = classNames(
+    styles.Stack,
+    vertical && styles.vertical,
+    spacing && styles[variationName('spacing', spacing)],
+    distribution && styles[variationName('distribution', distribution)],
+    alignment && styles[variationName('alignment', alignment)],
+    wrap === false && styles.noWrap,
+  );
+
+  const itemMarkup = elementChildren(children).map((child, index) => {
+    const props = {key: index};
+    return wrapWithComponent(child, Item, props);
+  });
+
+  return <div className={className}>{itemMarkup}</div>;
+}) as NamedExoticComponent<StackProps> & {
```

Comment:
In other cases where we need this recasting it's been done at the top of the constant declaration rather than using an `as`, see [Autocomplete for any example](https://github.com/Shopify/polaris-react/blob/master/src/components/Autocomplete/Autocomplete.tsx#L46). Can we keep to that style for consistency?


```js
export const Stack: React.NamedExoticComponent<StackProps> & {
  Item: typeof Item;
} = function memo(function Stack({
//...
```

Reply:
Typing is always preferred overcasting, however, unfortunately, we can't in this situation. 
```ts
Type 'NamedExoticComponent<StackProps>' is not assignable to type 'NamedExoticComponent<StackProps> & { Item: ({ children, fill }: ItemProps) => Element; }'.
  Property 'Item' is missing in type 'NamedExoticComponent<StackProps>' but required in type '{ Item: ({ children, fill }: ItemProps) => Element; }'
```

Reply:
well that's a little annoying :(

/me waves fist at React.memo